Server/ConcordiaCurriculumManager/Repositories/GroupRepository.cs

```diff
@@ -44,7 +45,7 @@ public async Task<bool> AddUserToGroup(Guid userId, Guid groupId)
     {
         var group = await _dbContext.Groups.FindAsync(groupId);
         var user = await _dbContext.Users.FindAsync(userId);
```

Comment:
Doesn't work, admins can still be added. I think it's because the way you query the user, there's no join with the Roles table. If you use the userRepository's GetUserById method to query for the user, it should work. Can you also write a test for this?